client/src/logic/impl/RoomSearchImpl.ts

```diff
@@ -0,0 +1,36 @@
+import { SearchNodeSuggestion } from "../../types/SearchNodeSuggestion";
+import { RoomSearch } from "../interfaces/RoomSearch";
+
+import { rooms } from "../../rooms";
+
+export class RoomSearchImpl implements RoomSearch {
+
+
+    sortedSuggestionsForStart(searchedText: string): SearchNodeSuggestion[] {
+        
+        // Mapping data into SearchNodeSuggestion type objects
+       const unsortedSuggestions : SearchNodeSuggestion[] = rooms.flatMap((room) => {
+            const { names } = room;
+            const destinationFilter = null;
+            return names.map((name) => new SearchNodeSuggestion(name, destinationFilter));
+        });
+
+        const filteredSuggestions: SearchNodeSuggestion[] = unsortedSuggestions.filter((suggestion) =>
+            suggestion.roomName.toLowerCase().startsWith(searchedText.toLowerCase()) || suggestion.roomName.toLowerCase().includes(searchedText.toLowerCase())
```

Comment:
`suggestion.roomName.toLowerCase().startsWith(searchedText.toLowerCase())` is redundant. The `includes` case already covers a case when a string starts with a substring. Remove the `startsWith` part and leave only the `includes` part of this condition.